plugins/drivers/utils/eventer.go

```diff
@@ -0,0 +1,101 @@
+package utils
+
+import (
+	"sync"
+	"time"
+
+	"github.com/hashicorp/nomad/plugins/drivers"
+	"golang.org/x/net/context"
+)
+
+var (
+	//DefaultSendEventTimeout is the timeout used when publishing events to consumers
+	DefaultSendEventTimeout = 2 * time.Second
+)
+
+// Eventer is a utility to control broadcast of TaskEvents to multiple consumers.
+// It also implements the TaskStats func in the DriverPlugin interface so that
+// it can be embedded in a implementing driver struct.
+type Eventer struct {
+	sync.RWMutex
```

Comment:
Only streamers needs guarding with a lock, so let's put this in a field and list it next to streamers.